contracts/libraries/PositionValue.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.7.0;
+
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+import '@uniswap/v3-core/contracts/libraries/FixedPoint128.sol';
+import '@uniswap/v3-core/contracts/libraries/TickMath.sol';
+import '@uniswap/v3-core/contracts/libraries/Tick.sol';
+import '../interfaces/INonfungiblePositionManager.sol';
+import './LiquidityAmounts.sol';
+import './PoolAddress.sol';
+import './PositionKey.sol';
+
+/// @title Returns information about the token value held in a Uniswap V3 NFT
+library PositionValue {
+    /// @notice returns the total token value including the principal and the fees owed
+    /// @param positionManager The Uniswap V3 NonfungiblePositionManager
+    /// @param tokenId The tokenId of the token for which to get the total value
+    /// @param sqrtRatioX96 The square root price X96 for which to calculate the principal amounts
+    /// @return amount0 The total amount of token0 including principal and fees
+    /// @return amount1 The total amount of token1 including principal and fees
+    function total(
+        INonfungiblePositionManager positionManager,
+        uint256 tokenId,
+        uint160 sqrtRatioX96
+    ) internal view returns (uint256 amount0, uint256 amount1) {
+        (PrincipalParams memory principalParams, FeeParams memory feeParams) =
+            _constructParams(positionManager, tokenId, sqrtRatioX96);
+
+        (amount0, amount1) = _principal(principalParams);
+
+        (uint256 amount0Fee, uint256 amount1Fee) = _fees(positionManager, feeParams);
+        amount0 += amount0Fee;
+        amount1 += amount1Fee;
+    }
+
+    struct PrincipalParams {
+        uint160 sqrtRatioX96;
+        int24 tickLower;
+        int24 tickUpper;
+        uint128 liquidity;
+    }
+
+    /// @notice Calculates the principal (currently acting as liquidity) owed to the token owner in the event
+    /// that the position is burned
+    /// @param positionManager The Uniswap V3 NonfungiblePositionManager
+    /// @param tokenId The tokenId of the token for which to get the total principal owed
+    /// @param sqrtRatioX96 The square root price X96 for which to calculate the principal amounts
```

Comment:
this is going to be hard to compute without additional guidance, we should create another library for going from some ratio (e.g. `1e6:1e18`, or `1 with 6 decimals : 1 with 18 decimals`) to sqrt ratio x96 after this pr